src/integrators/integrator_utils.jl

```diff
@@ -338,7 +338,9 @@ function handle_callbacks!(integrator)
   if integrator.u_modified
     handle_callback_modifiers!(integrator)
   end
-  integrator.u_modified = false
+  if !(typeof(integrator.alg) <: MultistepAlgorithms)
+    integrator.u_modified = false
```

Comment:
This one may be unnecessary because of later on anyways. I would try deleting it.

https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/53c8950acea4aaf46880fbe5329828d3ba010c97/src/integrators/integrator_utils.jl#L341-L343